src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -261,7 +261,15 @@ impl Cargo {
             }
         }
 
-        for arg in linker_args(builder, compiler.host, LldThreads::Yes) {
+        // When determining flags for the host (build scripts/proc macros),
+        // we use the snapshot compiler when building `Mode::Std` tools, and
+        // the current compiler when building anything else.
+        // We need to determine the current stage here to pass proper linker args (e.g. -C vs -Z)
+        // to the compiler used to compile build scripts.
+        // This should stay synchronized with the [cargo] function.
+        let host_stage = if mode == Mode::Std { 0 } else { compiler.stage };
```

Comment:
I think that this should be just `compiler.stage` now, as std is now compiled with the stage 1 compiler.

Reply:
Let's move this to zulip soon, because `compiler.stage` does not work.